tools/codegen/gen.py

```diff
@@ -231,41 +231,80 @@ def get_static_dispatch_backend(f: NativeFunction, backend_index: BackendIndex)
+    if backend_index is None or f.manual_kernel_registration:
+        return None
+    target_sig = CppSignatureGroup.from_native_function(f, method=False, fallback_binding=False).signature
+    name = target_sig.name()
+    exprs = translate(cpp_sig.arguments(), target_sig.arguments(), method=method)
+    exprs_str = ', '.join(a.expr for a in exprs)
+    if f.structured_delegate is not None:
+        # TODO: for ops with structured_delegate it should check the dispatch table of
+        # the out variant instead. For now, these structured ops all have CPU/CUDA kernels
+        # so we always dispatch to the `backend`, but this could be wrong when we
+        # migrate math/default_backend ops to use structured delegate.
+        if backend_index.has_kernel(f) or backend_index.dispatch_key in STRUCTURED_DISPATCH_KEYS:
+            return f'return at::{backend_index.dispatch_key.lower()}::{name}({exprs_str});'
+        else:
+            return f'TORCH_CHECK(false, "Static dispatch does not support {name} for {backend_index.dispatch_key}.");'
 
+    if backend_index.has_kernel(f):
+        return f'return at::{backend_index.dispatch_key.lower()}::{name}({exprs_str});'
+    elif f.has_composite_explicit_autograd_kernel:
+        return f'return at::{DispatchKey.CompositeExplicitAutograd.lower()}::{name}({exprs_str});'
+    elif f.has_composite_implicit_autograd_kernel:
+        return f'return at::{DispatchKey.CompositeImplicitAutograd.lower()}::{name}({exprs_str});'
+
+    return f'TORCH_CHECK(false, "Static dispatch does not support {name} for {backend_index.dispatch_key}.");'
 
 def static_dispatch(
         f: NativeFunction, cpp_sig: CppSignature,
-        *, method: bool, backend_index: Optional[BackendIndex]
+        *, method: bool, backend_indices: List[BackendIndex]
 ) -> Optional[str]:
-    if backend_index is None or f.manual_kernel_registration:
+    if backend_indices is None or f.manual_kernel_registration:
         return None
     target_sig = CppSignatureGroup.from_native_function(f, method=False, fallback_binding=False).signature
     name = target_sig.name()
     exprs = translate(cpp_sig.arguments(), target_sig.arguments(), method=method)
     exprs_str = ', '.join(a.expr for a in exprs)
 
-    dispatch_key = get_static_dispatch_backend(f, backend_index)
-    if dispatch_key is not None:
-        return f'return at::{dispatch_key.lower()}::{name}({exprs_str});'
-
-    return f'TORCH_CHECK(false, "Static dispatch does not support {name} for {backend_index.dispatch_key}.");'
+    keys = [b for b in backend_indices if b.has_kernel(f) or f.structured_delegate is not None]
+    if len(keys) == 1:
+        return generate_static_dispatch(f, cpp_sig, method=method, backend_index=keys[0])
+    elif len(keys) == 0:
+        return generate_static_dispatch(f, cpp_sig, method=method, backend_index=backend_indices[0])
+    else:
+        return f'TORCH_CHECK(false, "Static dispatch does not support backends with multiple kernels");'
```

Comment:
Hmm... if you pass in two backends to the codegen (say `CPU` and `QuantizedCPU`), then won't the codegen emit this torch check for every `f.structured_delegate` NativeFunction?

Reply:
The else part is implemented later in the stack https://github.com/pytorch/pytorch/pull/75607. Split them up to keep diff sizes manageable. 
Also if we pass in multiple backends for a native function, and if they have a structured_delegate and one or more kernels for the same, then this torch check will be emitted as of this diff.